gazebo_plugins/src/gazebo_ros_ackermann_drive.cpp

```diff
@@ -0,0 +1,592 @@
+      "Front right wheel joint [%s] not found, plugin will not work.", front_right_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto front_left_joint = _sdf->Get<std::string>("front_left_joint", "front_left_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::FRONT_LEFT] = _model->GetJoint(front_left_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::FRONT_LEFT]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Front left wheel joint [%s] not found, plugin will not work.", front_left_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto rear_right_joint = _sdf->Get<std::string>("rear_right_joint", "rear_right_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::REAR_RIGHT] = _model->GetJoint(rear_right_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::REAR_RIGHT]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Rear right wheel joint [%s] not found, plugin will not work.", rear_right_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto rear_left_joint = _sdf->Get<std::string>("rear_left_joint", "rear_left_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::REAR_LEFT] = _model->GetJoint(rear_left_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::REAR_LEFT]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Rear left wheel joint [%s] not found, plugin will not work.", rear_left_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto right_steering_joint =
+    _sdf->Get<std::string>("right_steering_joint", "right_steering_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_RIGHT] =
+    _model->GetJoint(right_steering_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_RIGHT]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Right wheel steering joint [%s] not found, plugin will not work.",
+      right_steering_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto left_steering_joint =
+    _sdf->Get<std::string>("left_steering_joint", "left_steering_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_LEFT] =
+    _model->GetJoint(left_steering_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_LEFT]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Left wheel steering joint [%s] not found, plugin will not work.",
+      left_steering_joint.c_str());
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  auto steering_wheel_joint =
+    _sdf->Get<std::string>("steering_wheel_joint", "steering_wheel_joint").first;
+  impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_WHEEL] =
+    _model->GetJoint(steering_wheel_joint);
+  if (!impl_->joints_[GazeboRosAckermannDrivePrivate::STEER_WHEEL]) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Steering wheel joint [%s] not found, plugin will not work.", steering_wheel_joint.c_str());
```

Comment:
How about making the steering wheel optional? I think most models may not have this level of detail. We'll need to be sure to check if it's not null below.